beacon-chain/rpc/apimiddleware/custom_hooks.go

```diff
@@ -91,3 +92,40 @@ func prepareGraffiti(endpoint gateway.Endpoint, _ http.ResponseWriter, _ *http.R
 	}
 	return nil
 }
+
+type tempSyncCommitteesResponseJson struct {
+	Data *tempSyncCommitteeValidatorsJson `json:"data"`
+}
+
+type tempSyncCommitteeValidatorsJson struct {
+	Validators          []string                              `json:"validators"`
+	ValidatorAggregates []*tempSyncSubcommitteeValidatorsJson `json:"validator_aggregates"`
+}
+
+type tempSyncSubcommitteeValidatorsJson struct {
+	Validators []string `json:"validators"`
+}
+
+// https://ethereum.github.io/beacon-APIs/?urls.primaryName=v2.0.0#/Beacon/getEpochSyncCommittees returns validator_aggregates as a nested array.
+// grpc-gateway returns a struct with nested fields which we have to transform into a plain 2D array.
+func prepareValidatorAggregates(body []byte, responseContainer interface{}) (bool, gateway.ErrorJson) {
```

Comment:
Is there a reason responseContainer is `interface{}` and not `*syncCommitteeResponseJson`?

Reply:
It's because the following code, which is in the main API processing function, is generic:

```
for _, hook := range endpoint.Hooks.OnPreDeserializeGrpcResponseBodyIntoContainer {
    ok, errJson := hook(grpcResponseBody, response)
```

`prepareValidatorAggregates` conforms to the signature of `hook`.